static/js/compose_closed_ui.js

```diff
@@ -64,17 +64,63 @@ function update_conversation_button(btn_text, title) {
     $("#left_bar_compose_private_button_big").prop("title", title);
 }
 
-function update_buttons(text_stream) {
+function update_buttons(text_stream, disable_reply, title_reply) {
     const title_stream = text_stream + " (c)";
     const text_conversation = $t({defaultMessage: "New private message"});
     const title_conversation = text_conversation + " (x)";
+    const text_deactivated_conversation = $t({defaultMessage: "Reply selected message"});
+    let title_deactivated_conversation = text_deactivated_conversation + " (r)";
+    if (narrow_state.pm_ids_string()) {
+        const user_ids = people.user_ids_string_to_ids_array(narrow_state.pm_ids_string());
+        const user_ids_dicts = user_ids.map((user_id) => ({id: user_id}));
+        if (people.verify_non_active_human_id(user_ids_dicts[0].id)) {
+            $(".compose_reply_button").attr("disabled", true);
+            title_deactivated_conversation = $t({
+                defaultMessage:
+                    "You are not allowed to send private messages in this organization.",
+            });
+            update_stream_button(text_stream, title_stream);
+            update_conversation_button(text_conversation, title_deactivated_conversation);
+            update_reply_button_deactivated_pm(disable_reply, title_reply);
+            return;
+        }
+
+        $(".compose_reply_button").attr("disabled", false);
+    }
     update_stream_button(text_stream, title_stream);
     update_conversation_button(text_conversation, title_conversation);
+    update_reply_button_deactivated_pm(disable_reply, title_reply);
+}
+
+// disable compose box reply button, when the recipient is a deactivated user.
+function update_reply_button_deactivated_pm(disable = false, title) {
+    $(".compose_reply_button").attr("disabled", disable);
+    if (!title) {
+        const title_text = $t({defaultMessage: "Reply to selected message"});
+        title = title_text + " (r)";
+    }
+    $(".compose_reply_button").prop("title", title);
 }
 
 export function update_buttons_for_private() {
     const text_stream = $t({defaultMessage: "New stream message"});
-    update_buttons(text_stream);
+    if (narrow_state.pm_ids_string()) {
+        const user_ids = people.user_ids_string_to_ids_array(narrow_state.pm_ids_string());
+        const user_ids_dicts = user_ids.map((user_id) => ({id: user_id}));
+        if (people.verify_non_active_human_id(user_ids_dicts[0].id)) {
+            // Disable compose box with a description when the recipient is a deactivated user.
+            $(".compose_reply_button").attr("disabled", true);
+            const disable_reply = true;
+            const title_reply = $t({
+                defaultMessage: "Cannot send message to a deactivated user.",
+            });
+            update_buttons(text_stream, disable_reply, title_reply);
+        } else {
+            update_buttons(text_stream);
+        }
```

Comment:
Is this a real case? Deactivated users shouldn't be able to login, that would defeat the purpose of them being deactivated.

Reply:
Yes, deactivated users are not allowed to log in, but in this case, an active user is trying to message a deactivated user and we are blocking the ability to send a message to a deactivated user.

- A active user is logged in.
- pm-with deactivated user.

Reply:
oh yeah!